source/backend/dal/Repositories/PropertyActivityRepository.cs

```diff
@@ -142,6 +160,29 @@ public bool TryDelete(long activityId)
             return deletedSuccessfully;
         }
 
+        /// <summary>
+        /// Delete an activity, and all property-activity relationships.
+        /// </summary>
+        /// <param name="activityId"></param>
+        /// <returns>Boolean of deletion sucess.</returns>
+        public bool TryDeleteFile(long activityId, long managementFileId)
+        {
+            // There may be zero to many of these relationships.
+            Context.PimsPropPropActivities.ForEach(pp =>
+            {
+                Context.PimsPropPropActivities.Remove(pp);
+            });
+
+            var propertyActivity = Context.PimsPropertyActivities.FirstOrDefault(x => x.PimsPropertyActivityId == activityId && x.ManagementFileId == managementFileId);
+            if(propertyActivity is null)
```

Comment:
Is this extra validation necessary? It would make sense to me if the repository deletes the activity with the given id, without considering the management file

Reply:
The idea with this repository method is that it should only be used on activities that were created with management files. This is because deleting a management file activity is different from deleting an ad-hoc activity (above).